File: leetsync/1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.go
func areAlmostEqual(s1 string, s2 string) bool {
	n := len(s1)
	diffCount := 0
	firstMismatch, secondMismatch := 0, 0
	for i := 0; i < n; i++ {
		if s1[i] != s2[i] {
			diffCount++
			if diffCount > 2 {
				return false
			} else if diffCount == 1 {
				firstMismatch = i
			} else if diffCount == 2 {
				secondMismatch = i
			}
		}
	}
	if s1[firstMismatch] == s2[secondMismatch] && s1[secondMismatch] == s2[firstMismatch] {
		return true
	}
	return false
}

File: leetsync/1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        if (s1.equals(s2)) {
            return true;
        }
        int n = s1.length();
        int diffCount = 0;
        int firstMismatch = 0, secondMismatch = 0;
        for (int i = 0; i < n; i++) {
            if (s1.charAt(i) != s2.charAt(i)) {
                diffCount++;
                if (diffCount > 2) {
                    return false;
                } else if (diffCount == 1) {
                    firstMismatch = i;
                } else if (diffCount == 2) {
                    secondMismatch = i;
                }
            }
        }
        if (s1.charAt(firstMismatch) == s2.charAt(secondMismatch)
                && s1.charAt(secondMismatch) == s2.charAt(firstMismatch)) {
            return true;
        }
        return false;
    }
}